### PR TITLE
WIP [cC]onfig => [cC]onfiguration

### PR DESCRIPTION
**Pull request by toshi0383**
- [x] Replace config to configuration. (Let me know if I'm going too far.😅)
- [x] Add deprecation message for old value usage.

Related: https://github.com/yonaskolb/XcodeGen/issues/59

### CODE REVIEW

File: Sources/XcodeGenKit/SettingsPresetFile.swift [hunk 1]
```diff
@@ -11,15 +11,15 @@ import xcodeproj
...
+    case configuration(ConfigurationType)
     case platform(Platform)
     case product(PBXProductType)
     case productPlatform(PBXProductType,Platform)
     case base
 
     var path: String {
         switch self {
-        case let .config(config): return "Configs/\(config.rawValue)"
+        case let .configuration(configuration): return "Configurations/\(configuration.rawValue)"
```
Comment (yonaskolb) [inline Sources/XcodeGenKit/SettingsPresetFile.swift]: This would break unless the presets `config` folder was renamed

File: CHANGELOG.md [hunk 2]
```diff
@@ -87,7 +87,7 @@
 - Added `carthageBuildPath` option [PR#34](https://github.com/yonaskolb/XcodeGen/pull/34)
 
 ### Fixed
-- Fixed installations of XcodeGen not applying build setting presets for configs, products, and platforms, due to missing resources
```
Comment (yonaskolb) [inline CHANGELOG.md]: references in the changelog shouldn't really be changed

**Comment by yonaskolb** [comment]
Oh wow, you literally renamed every single instance of config in the project :)
I must say, I think I prefer `config` in code. It's just much easier to read and type.

At the moment in terms of user facing properties in the spec we have following:
- `project.configs`
- `project.configFiles` (not added yet)
- `target.configs`
- `target.configFiles`
- `settings.configs` (in project, target and setting groups)

These should be the only things that change (as well as the documentation for them).

In addition, I'm actually still on the fence about this change. Does anyone else here have any opinions?

**Comment by toshi0383** [comment]
IMO `config` is very troublesome. New user might think `configFiles` is related to `config`.  Xcode or xcodebuild does not use `config` for shorthand name for `Configuration`. I think it's more user friendly to avoid potential confusion with xcconfig. Some user even don't know what `Configuration` is either, so we should be very very kind as possible.

> `Usage: xcodebuild [-project <projectname>] [[-target <targetname>]...|-alltargets] [-configuration <configurationname>] [-arch <architecture>]... [-sdk [<sdkname>|<sdkpath>]] [-showBuildSettings] [<buildsetting>=<value>]... [<buildaction>]...`

**Comment by toshi0383** [comment]
I'm not planning to work on this one for now. (I got used to the code-base and I feel good.👌)

**Pull request closed without merging**